server/src/main/java/org/candlepin/policy/js/compliance/ComplianceStatus.java

```diff
@@ -101,15 +101,15 @@ public void addNonCompliantProduct(String productId) {
 
     public void addPartiallyCompliantProduct(String productId, Entitlement entitlement) {
         if (!partiallyCompliantProducts.containsKey(productId)) {
-            partiallyCompliantProducts.put(productId, new HashSet<Entitlement>());
+            partiallyCompliantProducts.put(productId, new HashSet<>());
```

Comment:
Depending on version requirements, we may need to be careful using the diamond operator in contexts like this. I believe in Java7, this doesn't resolve properly/at all, so it may not work. In Java 8, using it like this on a generic method also fails, since it ends up resolving to Object. We're safe here since the method is not generic, but we may see some confusion when we see the same code structure fail in one place and not in another.